org.alloytools.alloy.application/src/main/java/templates/TestTemplateFile.vm

```diff
@@ -412,21 +469,61 @@ class CentralController:
 
     def __init__(self):
         self.concState = $!{rootState.getName()}_State(None, None)
-        self.commands = {"help": self.printHelp, "states": self.printStates, "execute": self.execute}
+        self.concState.default_initialize()
+        self.commands = {"help": self.printHelp, "execute": self.execute}
         #if(${allEnvEvents.size()}> 0)
         self.possibleEvents = {"${allEnvEvents.get(0).getModifiedName()}"#foreach($event in ${allEnvEvents.subList(1, ${allEncEvents.size()})}), "$!{event.getModifiedName()}"#end}
         #else
         self.possibleEvents = {}
         #end
-
+    
+    def handle_event(self):
+        unused_conc_states = self.concState.get_conc_descendants()
+        old_active_states = dict()
+        for conc_state in unused_conc_states:
+            old_active_states[conc_state.__repr__()] = conc_state.get_active_child_state_repr()
+        prev_num_unused_conc_states = -1
+        
+        while len(unused_conc_states) > 0 and not prev_num_unused_conc_states == len(unused_conc_states):
```

Comment:
`while len(unused_conc_states) > 0 and prev_num_unused_conc_states != len(unused_conc_states):`?